x-pack/otel/exporter/logstashexporter/internal/event.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/client"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+)
+
+func parseEvent(ctx context.Context, logRecord *plog.LogRecord) (beat.Event, error) {
```

Comment:
NIT: I think all the functions here are generic enough that would make sense to live in their own package, but I don't want to enforce that yet. Some of the parse functions could live in `beatconverter` and the context related one (GetBeatVersion) could live in a `otelctx` one.

Reply:
It makes sense, everything we pass down to the collector's context on the `otelconsumer` could be placed into a new package as you mentioned, so other components could also leverage it. 

Regarding the parsing function, do you think it would be useful for other components as well? The Logstash exporter is a kind of wrapper, that's why we're parsing `*plog.LogRecord` back to `beat.Event` here, so it can re-use the existing Logstash clients without the need to build a new one from scratch that supports pdata.

Reply:
I think we agree on the reasons for isolating the context package, but I cannot justify abstracting the parse changes right now since it is only used here. It should be fine to leave it as is for now.